app/questionnaire/questionnaire_schema.py

```diff
@@ -106,6 +116,8 @@ def __init__(
         )
         self._list_names_by_list_repeating_block_id: dict[str, str] = {}
         self._repeating_block_answer_ids: set[str] = set()
+        self._lists_populated_by_list_collector: set[str] = set()
+        self._schema_dependent_lists: set[str] = set()
```

Comment:
I'm not sure what `schema_dependent_lists` actually mean?. It seems to indicate more than it is. Maybe that is just me. We are already in the context of the schema, so can this just be like `list_names` or something similar?. The `dependent` bit confuses me and makes me think there is more to it?.

Reply:
See if it is needed after https://github.com/ONSdigital/eq-questionnaire-runner/pull/1198#discussion_r1317457814